### PR TITLE
[New Rule] Potential Priivilege Escalation via InstallerFileTakeOver

### LINKED ISSUE #1628: [New Rule] InstallerTakeOver CVE-2021-41379
## Description
Detects exploitation of CVE-2021-41379

### Target indexes
`logs-*, winlogbeat-*`

### Additional requirements
This should work with Elastic Agent by default, however you may need to tweak your sysmon config to audit file modifications and the rule syntax.

### Target Operating Systems
Windows

### Platforms
n/a

### Tested ECS Version
1.12

## Optional Info
Tested using https://github.com/klinix5/InstallerFileTakeOver - my hypothesis is that if the exploit needs to change "C:\Program Files (x86)\Microsoft\Edge\Application\96.0.1054.29\elevation_service.exe" then as long as we catch a change to that file and then a cmd.exe being run from a **now modified** elevation_service.exe it shouldn't be too noisy.

### Query
```
sequence with maxspan=30s
[file where file.name  == "elevation_service.exe" and event.action == "overwrite" ]
[process where process.name: "cmd.exe" and user.name: "SYSTEM" and process.parent.name: "elevation_service.exe" and event.action: "start" ]
```

### References
https://www.bleepingcomputer.com/news/microsoft/new-windows-zero-day-with-public-exploit-lets-you-become-an-admin/
https://www.logpoint.com/en/blog/detecting-privilege-escalation-zero-day-cve-2021-41379/
https://github.com/SigmaHQ/sigma/blob/db03d08b1105f33de763187da7681725ba70accd/rules/windows/builtin/win_vul_cve_2021_41379.yml
https://github.com/klinix5/InstallerFileTakeOver

## Example Data

<details>

```json
{
  "_id": "QX1uVX0B7tzLaFYRGB1g",
  "_index": ".ds-logs-endpoint.events.file-default-2021.10.29-000001",
  "_score": "1",
  "_type": "_doc",
  "@timestamp": "2021-11-25T04:49:52.814Z",
  "agent": {
    "id": "49004241-c3ea-426d-a92a-9db9f2a6351d",
    "type": "endpoint",
    "version": "7.15.1"
  },
  "data_stream": {
    "dataset": "endpoint.events.file",
    "namespace": "default",
    "type": "logs"
  },
  "ecs": {
    "version": "1.11.0"
  },
  "elastic": {
    "agent": {
      "id": "49004241-c3ea-426d-a92a-9db9f2a6351d"
    }
  },
  "event": {
    "action": "overwrite",
    "agent_id_status": "verified",
    "category": "file",
    "created": "2021-11-25T04:49:52.814Z",
    "dataset": "endpoint.events.file",
    "id": "MNvhVtD/9DMCwGzh+++++khc",
    "ingested": "2021-11-25T04:50:18.000Z",
    "kind": "event",
    "module": "endpoint",
    "sequence": "111887",
    "type": "change"
  },
  "file": {
    "Ext": {
      "entropy": "0",
      "header_data": "",
      "monotonic_id": "20743"
    },
    "extension": "exe",
    "name": "elevation_service.exe",
    "path": "C:\\Program Files (x86)\\Microsoft\\Edge\\Application\\96.0.1054.29\\elevation_service.exe",
    "size": "0"
  },
  "host": {
    "architecture": "x86_64",
    "hostname": "DESKTOP-T48PNO7",
    "id": "a369b334-f414-4d28-9b43-e49dfb4ef4c2",
    "ip": "192.168.254.138,fe80::7556:e75b:9027:8f2b,169.254.177.239,fe80::8496:c81c:aa06:b1ef,127.0.0.1,::1",
    "mac": "00:0c:29:70:e5:73,f0:18:98:5c:60:35",
    "name": "DESKTOP-T48PNO7",
    "os": {
      "Ext": {
        "variant": "Windows 10 Pro"
      },
      "family": "windows",
      "full": "Windows 10 Pro 2009 (10.0.19044.1288)",
      "kernel": "2009 (10.0.19044.1288)",
      "name": "Windows",
      "platform": "windows",
      "type": "windows",
      "version": "2009 (10.0.19044.1288)"
    }
  },
  "message": "Endpoint file event",
  "process": {
    "entity_id": "NDkwMDQyNDEtYzNlYS00MjZkLWE5MmEtOWRiOWYyYTYzNTFkLTg0MjAtMTMyODIyODkzODAuNzEzNDIzNTAw",
    "executable": "C:\\Users\\GerardSmith\\Desktop\\InstallerFileTakeOver.exe",
    "Ext": {
      "ancestry": "NDkwMDQyNDEtYzNlYS00MjZkLWE5MmEtOWRiOWYyYTYzNTFkLTUxNzYtMTMyODIxMTYyMDAuOTA1MzkxMTAw"
    },
    "name": "InstallerFileTakeOver.exe",
    "parent": {
      "pid": "5176"
    },
    "pid": "8420"
  },
  "user": {
    "domain": "AzureAD",
    "id": "S-1-12-1-3823846163-1128497860-3108885141-1004433371",
    "name": "GerardSmith"
  }
}
{
  "_id": "TH1uVX0B7tzLaFYRGB1g",
  "_index": ".ds-logs-endpoint.events.process-default-2021.10.29-000001",
  "_score": "1",
  "_type": "_doc",
  "@timestamp": "2021-11-25T04:49:53.166Z",
  "agent": {
    "id": "49004241-c3ea-426d-a92a-9db9f2a6351d",
    "type": "endpoint",
    "version": "7.15.1"
  },
  "data_stream": {
    "dataset": "endpoint.events.process",
    "namespace": "default",
    "type": "logs"
  },
  "ecs": {
    "version": "1.11.0"
  },
  "elastic": {
    "agent": {
      "id": "49004241-c3ea-426d-a92a-9db9f2a6351d"
    }
  },
  "event": {
    "action": "start",
    "agent_id_status": "verified",
    "category": "process",
    "created": "2021-11-25T04:49:53.166Z",
    "dataset": "endpoint.events.process",
    "id": "MNvhVtD/9DMCwGzh+++++kjA",
    "ingested": "2021-11-25T04:50:18.000Z",
    "kind": "event",
    "module": "endpoint",
    "sequence": "111930",
    "type": "start"
  },
  "host": {
    "architecture": "x86_64",
    "hostname": "DESKTOP-T48PNO7",
    "id": "a369b334-f414-4d28-9b43-e49dfb4ef4c2",
    "ip": "192.168.254.138,fe80::7556:e75b:9027:8f2b,169.254.177.239,fe80::8496:c81c:aa06:b1ef,127.0.0.1,::1",
    "mac": "00:0c:29:70:e5:73,f0:18:98:5c:60:35",
    "name": "DESKTOP-T48PNO7",
    "os": {
      "Ext": {
        "variant": "Windows 10 Pro"
      },
      "family": "windows",
      "full": "Windows 10 Pro 2009 (10.0.19044.1288)",
      "kernel": "2009 (10.0.19044.1288)",
      "name": "Windows",
      "platform": "windows",
      "type": "windows",
      "version": "2009 (10.0.19044.1288)"
    }
  },
  "message": "Endpoint process event",
  "process": {
    "args": "C:\\Windows\\system32\\cmd.exe",
    "args_count": "1",
    "code_signature": {
      "exists": "true",
      "status": "trusted",
      "subject_name": "Microsoft Windows",
      "trusted": "true"
    },
    "command_line": "\"C:\\Windows\\system32\\cmd.exe\"",
    "entity_id": "NDkwMDQyNDEtYzNlYS00MjZkLWE5MmEtOWRiOWYyYTYzNTFkLTUwNDgtMTMyODIyODkzOTMuMTY2Nzk0NjAw",
    "executable": "C:\\Windows\\SysWOW64\\cmd.exe",
    "Ext": {
      "ancestry": "NDkwMDQyNDEtYzNlYS00MjZkLWE5MmEtOWRiOWYyYTYzNTFkLTM1NDAtMTMyODIyODkzOTMuNzAzMDcwMA==,NDkwMDQyNDEtYzNlYS00MjZkLWE5MmEtOWRiOWYyYTYzNTFkLTk3OTYtMTMyODIyODkzOTIuODQ0NDEyNDAw,NDkwMDQyNDEtYzNlYS00MjZkLWE5MmEtOWRiOWYyYTYzNTFkLTY0NC0xMzI4MjExNjE0My4yMDk3NzE1MDA=,NDkwMDQyNDEtYzNlYS00MjZkLWE5MmEtOWRiOWYyYTYzNTFkLTQ5Mi0xMzI4MjExNjE0Mi45NDg0MjM4MDA=",
      "authentication_id": "0x3e7",
      "code_signature": "{\"trusted\":true,\"subject_name\":\"Microsoft Windows\",\"exists\":true,\"status\":\"trusted\"}",
      "token": {
        "elevation_level": "default",
        "integrity_level_name": "system"
      }
    },
    "hash": {
      "md5": "d0fce3afa6aa1d58ce9fa336cc2b675b",
      "sha1": "4048488de6ba4bfef9edf103755519f1f762668f",
      "sha256": "4d89fc34d5f0f9babd022271c585a9477bf41e834e46b991deaa0530fdb25e22"
    },
    "name": "cmd.exe",
    "parent": {
      "args": "C:\\Program Files (x86)\\Microsoft\\Edge\\Application\\96.0.1054.29\\elevation_service.exe,/svc",
      "args_count": "2",
      "command_line": "\"C:\\Program Files (x86)\\Microsoft\\Edge\\Application\\96.0.1054.29\\elevation_service.exe\" /svc",
      "entity_id": "NDkwMDQyNDEtYzNlYS00MjZkLWE5MmEtOWRiOWYyYTYzNTFkLTM1NDAtMTMyODIyODkzOTMuNzAzMDcwMA==",
      "executable": "C:\\Program Files (x86)\\Microsoft\\Edge\\Application\\96.0.1054.29\\elevation_service.exe",
      "name": "elevation_service.exe",
      "pid": "3540"
    },
    "pe": {
      "original_file_name": "Cmd.Exe"
    },
    "pid": "5048",
    "working_directory": "C:\\Windows\\system32\\"
  },
  "user": {
    "domain": "NT AUTHORITY",
    "id": "S-1-5-18",
    "name": "SYSTEM"
  }
}
```

</details>

### PR DESCRIPTION
**Pull request by w0rk3r**
## Issues

Resolves #1628 

## Summary

Identifies a potential exploitation of InstallerTakeOver/CVE-2021-41379 default PoC execution. Successful exploitation
allows an unprivileged user to escalate privileges to SYSTEM.

Rule Scope: This rule detects the default execution of the PoC, which overwrites the elevation_service.exe DACL and copy itself to the location to escalate privileges. Still, an attacker could take over any file that is not in use (locked), such scenarios are not covered in this rule.

### CODE REVIEW

File: rules/windows/privilege_escalation_installertakeover.toml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+author = ["Elastic"]
+description = """
+Identifies a potential exploitation of InstallerTakeOver/CVE-2021-41379 default PoC execution. Successful exploitation
+allows an unprivileged user to escalate privileges to SYSTEM.
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Priivilege Escalation via InstallerFileTakeOver"
```
Comment (Samirbous) [inline rules/windows/privilege_escalation_installertakeover.toml:16]:
```suggestion
name = "Potential Privilege Escalation via InstallerFileTakeOver"
```

**Review by Samirbous — APPROVED** [review]
👍

**Comment by hilt86** [comment]
I received a type conflict on user.id - if you have an osquery result with user.id the type can conflict. Here is the actual error : 

```
verification_exception: Found 1 problem line 2:5: Cannot use field [user.id] due to ambiguities being mapped as [2] incompatible types: [keyword] in [.ds-logs-endpoint.alerts-default-2021.10.20-000004, .ds-logs-endpoint.alerts-default-2021.10.27-000005, .ds-logs-endpoint.alerts-default-2021.11.03-000006, .ds-logs-endpoint.alerts-default-2021.11.10-000007, .ds-logs-endpoint.alerts-default-2021.11.17-000008, .ds-logs-endpoint.alerts-default-2021.11.24-000009, .ds-logs-endpoint.events.file-default-2021.10.19-000005, .ds-logs-endpoint.events.file-default-2021.10.26-000006, .ds-logs-endpoint.events.file-default-2021.11.02-000007, .ds-logs-endpoint.events.file-default-2021.11.09-000008, .ds-logs-endpoint.events.file-default-2021.11.16-000009, .ds-logs-endpoint.events.file-default-2021.11.23-000010, .ds-logs-endpoint.events.library-default-2021.10.19-000004, .ds-logs-endpoint.events.library-default-2021.10.26-000005, .ds-logs-endpoint.events.library-default-2021.11.02-000006, .ds-logs-endpoint.events.library-default-2021.11.09-000007, .ds-logs-endpoint.events.library-default-2021.11.16-000008, .ds-logs-endpoint.events.library-default-2021.11.23-000009, .ds-logs-endpoint.events.network-default-2021.10.19-000004, .ds-logs-endpoint.events.network-default-2021.10.26-000005, .ds-logs-endpoint.events.network-default-2021.11.02-000006, .ds-logs-endpoint.events.network-default-2021.11.09-000007, .ds-logs-endpoint.events.network-default-2021.11.16-000008, .ds-logs-endpoint.events.network-default-2021.11.23-000009, .ds-logs-endpoint.events.process-default-2021.10.19-000004, .ds-logs-endpoint.events.process-default-2021.10.26-000005, .ds-logs-endpoint.events.process-default-2021.11.02-000006, .ds-logs-endpoint.events.process-default-2021.11.09-000007, .ds-logs-endpoint.events.process-default-2021.11.16-000008, .ds-logs-endpoint.events.process-default-2021.11.23-000009, .ds-logs-endpoint.events.registry-default-2021.10.19-000004, .ds-logs-endpoint.events.registry-default-2021.10.26-000005, .ds-logs-endpoint.events.registry-default-2021.11.02-000006, .ds-logs-endpoint.events.registry-default-2021.11.09-000007, .ds-logs-endpoint.events.registry-default-2021.11.16-000008, .ds-logs-endpoint.events.registry-default-2021.11.23-000009, .ds-logs-endpoint.events.security-default-2021.10.19-000004, .ds-logs-endpoint.events.security-default-2021.10.26-000005, .ds-logs-endpoint.events.security-default-2021.11.02-000006, .ds-logs-endpoint.events.security-default-2021.11.09-000007, .ds-logs-endpoint.events.security-default-2021.11.16-000008, .ds-logs-endpoint.events.security-default-2021.11.23-000009, .ds-logs-system.auth-default-2021.10.20-000006, .ds-logs-system.auth-default-2021.10.27-000007, .ds-logs-system.auth-default-2021.11.03-000008, .ds-logs-system.auth-default-2021.11.10-000009, .ds-logs-system.auth-default-2021.11.17-000010, .ds-logs-system.auth-default-2021.11.24-000011, .ds-logs-system.auth-usbase-2021.10.20-000003, .ds-logs-system.auth-usbase-2021.10.27-000004, .ds-logs-system.auth-usbase-2021.11.03-000005, .ds-logs-system.auth-usbase-2021.11.10-000006, .ds-logs-system.auth-usbase-2021.11.17-000007, .ds-logs-system.auth-usbase-2021.11.24-000008, .ds-logs-system.security-default-2021.10.19-000005, .ds-logs-system.security-default-2021.10.26-000006, .ds-logs-system.security-default-2021.11.02-000007, .ds-logs-system.security-default-2021.11.09-000008, .ds-logs-system.security-default-2021.11.11-000009, .ds-logs-system.security-default-2021.11.18-000010, .ds-logs-system.security-default-2021.11.25-000011, partial-.ds-logs-endpoint.alerts-default-2021.09.27-000001, partial-.ds-logs-endpoint.alerts-default-2021.10.19-000002, partial-.ds-logs-endpoint.alerts-default-2021.10.20-000003, partial-.ds-logs-endpoint.events.file-default-2021.08.09-000002, partial-.ds-logs-endpoint.events.file-default-2021.09.08-000003, partial-.ds-logs-endpoint.events.file-default-2021.10.08-000004, partial-.ds-logs-endpoint.events.library-default-2021.07.22-000001, partial-.ds-logs-endpoint.events.library-default-2021.08.21-000002, partial-.ds-logs-endpoint.events.library-default-2021.09.20-000003, partial-.ds-logs-endpoint.events.network-default-2021.07.22-000001, partial-.ds-logs-endpoint.events.network-default-2021.08.21-000002, partial-.ds-logs-endpoint.events.network-default-2021.09.20-000003, partial-.ds-logs-endpoint.events.process-default-2021.07.22-000001, partial-.ds-logs-endpoint.events.process-default-2021.08.21-000002, partial-.ds-logs-endpoint.events.process-default-2021.09.20-000003, partial-.ds-logs-endpoint.events.registry-default-2021.07.22-000001, partial-.ds-logs-endpoint.events.registry-default-2021.08.21-000002, partial-.ds-logs-endpoint.events.registry-default-2021.09.20-000003, partial-.ds-logs-endpoint.events.security-default-2021.07.22-000001, partial-.ds-logs-endpoint.events.security-default-2021.08.21-000002, partial-.ds-logs-endpoint.events.security-default-2021.09.20-000003, partial-.ds-logs-system.auth-default-2021.07.22-000001, partial-.ds-logs-system.auth-default-2021.08.06-000002, partial-.ds-logs-system.auth-default-2021.09.05-000003, partial-.ds-logs-system.auth-default-2021.10.05-000004, partial-.ds-logs-system.auth-default-2021.10.13-000005, partial-.ds-logs-system.auth-usbase-2021.09.22-000001, partial-.ds-logs-system.auth-usbase-2021.10.13-000002, partial-.ds-logs-system.security-default-2021.07.22-000001, partial-.ds-logs-system.security-default-2021.08.06-000002, partial-.ds-logs-system.security-default-2021.09.05-000003, partial-.ds-logs-system.security-default-2021.10.05-000004], [long] in [.ds-logs-osquery_manager.result-default-2021.11.09-000001, .ds-logs-osquery_manager.result-default-2021.11.16-000002, .ds-logs-osquery_manager.result-default-2021.11.23-000003]
```

**Comment by w0rk3r** [comment]
@hilt86 the defined indexes are the same as the ones in this PR?

File: rules/windows/privilege_escalation_installertakeover.toml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+rule_id = "58c6d58b-a0d3-412d-b3b8-0981a9400607"
+severity = "high"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Privilege Escalation"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+/* This rule is compatible with both Sysmon and Elastic Endpoint */
+
+process where event.action : ("start", "Process Create*") and
```
Comment (brokensound77) [inline rules/windows/privilege_escalation_installertakeover.toml:60]:
```suggestion
process where event.type == "start" and 
```

File: rules/windows/privilege_escalation_installertakeover.toml [hunk 3]
```diff
@@ -0,0 +1,83 @@
+[metadata]
+creation_date = "2021/11/25"
+maturity = "production"
+updated_date = "2021/11/25"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies a potential exploitation of InstallerTakeOver/CVE-2021-41379 default PoC execution. Successful exploitation
```
Comment (brokensound77) [inline rules/windows/privilege_escalation_installertakeover.toml:9]:
```suggestion
Identifies a potential exploitation of InstallerTakeOver (CVE-2021-41379) default PoC execution. Successful exploitation
```

File: rules/windows/privilege_escalation_installertakeover.toml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+license = "Elastic License v2"
+name = "Potential Privilege Escalation via InstallerFileTakeOver"
+note = """## Triage and analysis.
+
+### Investigating Potential Priivilege Escalation via InstallerFileTakeOver
+
+InstallerFileTakeOver is a weaponized EoP PoC to the CVE-2021-41379 vulnerability. Upon successful exploitation,
+an unprivileged user will escalate privileges to SYSTEM/NT AUTHORITY.
+
+This rule detects the default execution of the PoC, which overwrites the elevation_service.exe DACL and copy itself
```
Comment (brokensound77) [inline rules/windows/privilege_escalation_installertakeover.toml:24]:
```suggestion
This rule detects the default execution of the PoC, which overwrites the `elevation_service.exe` DACL and copy itself
```

File: rules/windows/privilege_escalation_installertakeover.toml [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+to the location to escalate privileges. Still, an attacker could take over any file that is not in use (locked), such
+scenarios are not covered in this rule.
```
Comment (brokensound77) [inline rules/windows/privilege_escalation_installertakeover.toml:26]:
```suggestion
to the location to escalate privileges. An attacker is able to still take over any file that is not in use (locked), which is outside the scope of this rule.
```

File: rules/windows/privilege_escalation_installertakeover.toml [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+type = "eql"
+
+query = '''
+/* This rule is compatible with both Sysmon and Elastic Endpoint */
+
+process where event.type == "start" and 
+    user.id : "S-1-5-18" and
+    (
+      (process.name : "elevation_service.exe" and 
+       not process.pe.original_file_name : "elevation_service.exe") or
```
Comment (brokensound77) [inline rules/windows/privilege_escalation_installertakeover.toml:64]:
```suggestion
       not process.pe.original_file_name == "elevation_service.exe") or
```

**Review by brokensound77 — APPROVED** [review]
✔️

**Pull request merged**